src/main/java/keycloak/scim_user_spi/SCIMUserStorageProvider.java

```diff
@@ -158,9 +157,17 @@ public boolean isValid(RealmModel realm, UserModel user, CredentialInput input)
 		if (!supportsCredentialType(input.getType()) || !(input instanceof UserCredentialModel)) return false;
 
 		UserCredentialModel cred = (UserCredentialModel)input;
-		String password = properties.getProperty(user.getUsername());
-		if (password == null) return false;
-		return password.equals(cred.getValue());
+		/* The password can either be validated locally in keycloak (tried first)
+		   or in the SCIM server */
+		if (session.userCredentialManager().isConfiguredLocally(realm, user, input.getType())) {
+			logger.debug("Local password validation for " + user.getUsername());
```

Comment:
Could you change the `logger.debug` calls to `logger.debugv`
~~~
logger.debugv("Local password validation for {0}", user.getUsername());
~~~

Reply:
Done, thanks for the suggestion.